tikv-client/src/main/java/com/pingcap/tikv/types/DecimalType.java

```diff
@@ -83,19 +85,48 @@ protected Object doConvertToTiDBType(Object value)
     } else {
       throw new ConvertNotSupportException(value.getClass().getName(), this.getClass().getName());
     }
-    return result;
+
+    int length = (int) this.getLength();
+    int decimal = this.getDecimal();
+    return toGivenPrecisionAndFrac(result, length, decimal);
+  }
+
+  private MyDecimal toGivenPrecisionAndFrac(java.math.BigDecimal value, int precision, int frac)
```

Comment:
can we add more comments for this method?

Reply:
sure

Reply:
done